flink-ml-benchmark/src/main/java/org/apache/flink/ml/benchmark/datagenerator/common/LabeledPointWithWeightGenerator.java

```diff
@@ -0,0 +1,194 @@
+
+    public static final Param<Integer> LABEL_ARITY =
+            new IntParam(
+                    "labelArity",
+                    "Arity of label. "
+                            + "If set to positive value, the label would be an integer in range [0, arity - 1]. "
+                            + "If set to zero, the label would be a continuous double in range [0, 1).",
+                    2,
+                    ParamValidators.gtEq(0));
+
+    public static final Param<Integer> WEIGHT_ARITY =
+            new IntParam(
+                    "weightArity",
+                    "Arity of weight. "
+                            + "If set to positive value, the weight would be an integer in range [1, arity]. "
+                            + "If set to zero, weight would be a continuous double in range [0, 1).",
+                    1,
+                    ParamValidators.gtEq(0));
+
+    private final Map<Param<?>, Object> paramMap = new HashMap<>();
+
+    public LabeledPointWithWeightGenerator() {
+        ParamUtils.initializeMapWithDefaultValues(paramMap, this);
+    }
+
+    public int getFeatureArity() {
+        return get(FEATURE_ARITY);
+    }
+
+    public LabeledPointWithWeightGenerator setFeatureArity(int value) {
+        return set(FEATURE_ARITY, value);
+    }
+
+    public int getLabelArity() {
+        return get(LABEL_ARITY);
+    }
+
+    public LabeledPointWithWeightGenerator setLabelArity(int value) {
+        return set(LABEL_ARITY, value);
+    }
+
+    public int getWeightArity() {
+        return get(WEIGHT_ARITY);
+    }
+
+    public LabeledPointWithWeightGenerator setWeightArity(int value) {
+        return set(WEIGHT_ARITY, value);
+    }
+
+    @Override
+    public Table[] getData(StreamTableEnvironment tEnv) {
+        StreamExecutionEnvironment env = TableUtils.getExecutionEnvironment(tEnv);
+
+        DataStream<Row> dataStream =
+                env.fromParallelCollection(
+                                new NumberSequenceIterator(1L, getNumValues()),
+                                BasicTypeInfo.LONG_TYPE_INFO)
+                        .map(
+                                new RandomLabeledPointWithWeightGenerator(
+                                        getSeed(),
+                                        getVectorDim(),
+                                        getFeatureArity(),
+                                        getLabelArity(),
+                                        getWeightArity()),
+                                new RowTypeInfo(
+                                        new TypeInformation[] {
+                                            DenseVectorTypeInfo.INSTANCE, Types.DOUBLE, Types.DOUBLE
+                                        },
+                                        new String[] {
+                                            getFeaturesCol(), getLabelCol(), getWeightCol()
+                                        }));
+
+        Table dataTable = tEnv.fromDataStream(dataStream);
```

Comment:
Should this class handle the `colNames` parameter?

Currently `DenseVectorArrayGenerator` checks that `getColNames().length > 0` when this parameter value is not null. Would it be better to verify `getColNames().length == 1` in this case? And we probably also need to take care of the case where the string contains `,` since this is a common-separated string.

Can you update `DenseVectorGenerator` as well for consistency?

I am also wondering if it would be more intuitive to update the description of `colNames` to be `An array of common-separated strings representing field names of the generated tables`.

Reply:
I agree that `LabeledPointWithWeightGenerator` should handle colNames, instead of featuresCol, labelCol and weightCol. I also think it is a good idea to verify `getColNames().length` in these table generators.

It has not been Flink ML's practice to handle potential column names that contain commas. Existing parameters, like `HasInputCols`, distinct column names from that they are different strings in the array. If `colNames = ["features,weight,label"]`, there would be ambiguity in what each element in the array stands for.

Still, I agree that we should handle situations when there are non-common characters in column name strings. I think we can add a ParamValidator that checks for the grammar of these strings, and apply this validator to classes like `HasColNames` and `HasInputCols`. How do you like this idea?